### DISCUSSION TITLE
replacing sharpdx with vortice.windows

### DISCUSSION THREAD

**Discussion by sjoerdev** [amerkoleci/Vortice.Windows#465] [General]
My name is Sjoerd. I'm a c# game/engine developer.
I have several old game projects that use SharpDX and WinForms.
I want to upgrade these old projects from .net framework to modern .net8.
I am looking for a drop in replacement for sharpdx for modern .net core.
I dont know where to find information on how to upgrade sharpdx projects to vortice.windows.
In case you want to have a discussion, this is my discord handle is "sjoerdev"

**Comment by amerkoleci** [comment]
Hi Sjoerd,
Thanks for reaching out.

Please don't open issues for questions when there are discussions, helping others in their migration in out of scope in this moment as my time is quite limited. 

Feel free to ask here if you have any specific questions. 

Amer

**Comment by h1cks** [comment]
I've done a full conversion from Sharpdx to Vortice earlier (I still use Sharpdx on my current game.).  But safe to say, that Vortice is alot closer to a native implementation, there are a lot of helper classes that Sharpdx  employed which quite frankly made it harder on you.  There are also challenges that the parameter order for in and out for some functions is reversed.  That is going to be your biggest challenge.

You could add inline static wrappers for all the Sharpdx calls, which I believe will cut down on conversion issues, sort of like a conversion to an intermediate state.   This would be the code you could lean into for your swap over.

Of note: Pay attention to alot of the GPU memory calls, im sure almost all of them are out of order.

There is an FMV thread for example in this discussion session I created a while ago, that demos many differences.

Good luck with the conversion.

**Comment by sjoerdev** [comment]
@h1cks do you have discord? i would like to ask you questions there about the sharpdx to vortice port.

**Comment by h1cks** [comment]
my discord id is ErnieDingo#6495

**Reply by sjoerdev** [reply]
i invited you as friend. did it work? my discord handle is: "sjoerdev"

**Comment by h1cks** [comment]
Hi. Well check when I return from overseas on weekend.
